src/melee.cpp

```diff
@@ -110,6 +110,56 @@ bool player::handle_melee_wear()
     return true;
 }
 
+bool player::handle_shields_wear()
```

Comment:
This doesn't need an entire extra function. Just change the `handle_melee_wear` function to allow taking an item reference argument and overload it to supply `weapon` as the default argument.